ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -60,8 +89,13 @@ final class Authorizer(now: () => Instant, ledgerId: String, participantId: Stri
       parties: Iterable[String],
       call: Req => Future[Res]): Req => Future[Res] =
     authorize(call) { claims =>
-      valid(claims) &&
-      parties.forall(claims.canReadAs)
+      for {
+        _ <- valid(claims)
+        _ <- parties.foldLeft[Either[AuthorizationError, Unit]](Right(()))((acc, e) =>
+          acc.flatMap(_ => claims.canReadAs(e)))
```

Comment:
This is quite long, it would be probably better to factor it out in a private method. Looking at the previous method, it appears that you can factor out the folding and pass the function that actually performs the check (something like `forall`, but on an `Either` -- not sure whether something like this exists already).